### DISCUSSION TITLE
How to filter by related (WP_Meta_Query analog)?

### DISCUSSION THREAD

**Discussion by bridge2web** [directus/directus#2839] [Archive]
I have a legacy Yii project. The db structure is adopted from WordPress (WP_Meta_Query).
I want to port this to Directus.

**tbl_post**
id
title
content
etc

**tbl_meta**
id
post_id
meta_key
meta_value

How to SELECT post by meta, for example:
tbl_meta.meta_key = 'room' AND tbl_meta.meta_value = 2
AND
tbl_meta.meta_key = 'floor' AND tbl_meta.meta_value = 3
AND
...
In Yii, I have grouped meta pairs into INNER JOINs.
Is it possible to do this with the Directus API?

**Comment by rijkvanzanten** [comment]
See the filter parameter: https://docs.directus.io/api/query/filter.html

**Reply by bridge2web** [reply]
Of course I read the documentation before asking the question.
But, it seems that the standard API cannot solve this problem, even in the simple case with one "meta query":
`/items/tbl_post?filter[tbl_meta.meta_key][eq]=room&filter[tbl_meta.meta_value][eq]=2`
The result is not correct... 
I suppose, that posts are returned that have any meta with meta_key = room, and with any meta with meta_value = 2, separately.
How to filter posts with meta tbl_meta.meta_key = 'room' AND tbl_meta.meta_value = 2 together?
